### PR TITLE
Redo code_sample shortcode

### PR DESCRIPTION
**Pull request by sftim**
Refactor shortcodes:
- `code_sample`
- `codenew` (legacy)
- `code` (legacy)

and make the text localizable. The new approach uses Hugo Pipes to get the image, allowing further improvements such as image optimization at build time.

Also, add some padding around the icon (~second~ third commit).

_[Example](https://k8s.io/docs/concepts/policy/limit-range/#limitrange-and-admission-checks-for-pods)_ vs _[preview](https://deploy-preview-46362--kubernetes-io-main-staging.netlify.app/docs/concepts/policy/limit-range/#limitrange-and-admission-checks-for-pods)_

/area web-development

This paves an easier path for [right to left locale support](https://github.com/kubernetes/website/issues/22730).

### CODE REVIEW

File: layouts/shortcodes/code_sample.html [hunk 1]
```diff
@@ -18,15 +18,16 @@
...
-<div class="highlight">
-    <div class="copy-code-icon" style="text-align:right">
+<div class="highlight code-sample">
+    <div class="copy-code-icon">
     {{ with $ghlink }}<a href="{{ . }}" download="{{ $file }}">{{ end }}<code>{{ $file }}</code>
     {{ if $ghlink }}</a>{{ end }}
-    <img src="{{ "images/copycode.svg" | relURL }}" style="max-height:24px; cursor: pointer" onclick="copyCode('{{ $file | anchorize }}')" title="Copy {{ $file }} to clipboard">
-    </img>
+    {{- with resources.Get "images/copycode.svg" -}}
+    <img src="{{  .RelPermalink }}" class="icon-copycode" onclick="copyCode('{{ $file | anchorize }}')" title="Copy {{ $file }} to clipboard"></img>
```
Comment (sftim) [inline layouts/shortcodes/code_sample.html:26]: We could / should localize this text.

**Review by dipesh-rawat — COMMENTED** [review]
Nice improvements! Positive changes with Hugo pipes for images, localized strings, CSS improvement and potential RTL support in the future. Also, the changes render as expected in the [preview page.](https://deploy-preview-46362--kubernetes-io-main-staging.netlify.app/docs/concepts/policy/limit-range/#limitrange-and-admission-checks-for-pods)

/lgtm

**Pull request merged**